File: atcoder/abc144B_81.nim
# uninhm
# https://atcoder.jp/contests/abc144/tasks/abc144_b
# implementation

from strutils import parseInt

let n = stdin.readLine.parseInt

var res: float
for i in 1 .. 9:
  res = n / i
  if res == res.int.float and
      res <= 9 and res >= 1:
    echo "Yes"
    quit()
echo "No"

File: atcoder/abc144B_81.ml
(* vicfred
 * https://atcoder.jp/contests/abc144/tasks/abc144_b
 * implementation
 * *)

let main =
    Scanf.scanf "%d" @@ fun n ->
      print_endline @@ if
        Array.fold_left (Array.fold_left ( || )) false @@
        Array.init 9 @@ fun i ->
          Array.init 9 @@ fun j -> n = (i+1)*(j+1)
    then "Yes"
    else "No"

